### PR TITLE
make binary search faster, add wheelchair and bike to departure lookup

### PR DESCRIPTION
**Pull request by kevinkreiser**
instead of spooling back to the beginning to get the first matching record ive slightly modified the 4 binary searches to find the first matching record. ive also simplified the logic after each one for the various other criteria.

in addition to that when asking for the next departure you can also specify that you need one that supports bike and/or wheelchair

### CODE REVIEW

File: valhalla/baldr/graphtile.h [hunk 1]
```diff
@@ -196,14 +196,18 @@ class GraphTile {
...
    * @return  Returns a pointer to the transit departure information.
    *          Returns nullptr if no departures are found.
    */
   const TransitDeparture* GetNextDeparture(const uint32_t lineid,
                                            const uint32_t current_time,
                                            const uint32_t day,
                                            const uint32_t dow,
-                                           bool  date_before_tile) const;
+                                           bool  date_before_tile,
+                                           bool wheelchair = false,
```
Comment (gknisely) [inline valhalla/baldr/graphtile.h]: no defaults needed.

Comment (kevinkreiser) [inline valhalla/baldr/graphtile.h]: i only did it so that i wouldnt break the build. do you want me to remove it?

Comment (gknisely) [inline valhalla/baldr/graphtile.h]: I will always set it, but up to you....if you leave make both true.

**Comment by dnesbitt61** [comment]
:shipit:

**Pull request merged**